src/@types/index.d.ts

```diff
@@ -0,0 +1,7 @@
+interface Client {
```

Comment:
Why global types here (`d.ts`)? Importing types makes it a bit easier to refactor and globals don't clash as easy.

Reply:
This file does/will contain the interface that we export to the global `window.FASTLY` on the host page and if we ever expose as a module via npm to load into other applications. Therefore it made sense for it to be global.